### PR TITLE
Add ScatterND FusedOp

### PR DESCRIPTION
**Pull request by Ewa21**

### CODE REVIEW

File: src/ngraph/op/fused/scatter_nd.hpp [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+#include "ngraph/op/op.hpp"
+
+namespace ngraph
+{
+    namespace op
+    {
+        namespace v0
+        {
+            /// \brief  Replace values within provided tensor by `updates` according to `indices`.
+            class NGRAPH_API ScatterND : public Op
```
Comment (tomdol) [inline src/ngraph/op/fused/scatter_nd.hpp]: Fused ops should inherit from `FusedOp`

File: src/ngraph/op/fused/scatter_nd.cpp [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+constexpr NodeTypeInfo op::ScatterND::type_info;
+
+shared_ptr<Node> op::ScatterND::copy_with_new_args(const NodeVector& new_args) const
+{
+    check_new_args_count(this, new_args);
+    return make_shared<ScatterND>(new_args.at(DATA), new_args.at(INDICES), new_args.at(UPDATES));
+}
+
+void op::ScatterND::validate_and_infer_types()
```
Comment (tomdol) [inline src/ngraph/op/fused/scatter_nd.cpp]: When you switch to `FusedOp` as a base class, this method should be renamed to `pre_validate_and_infer_types`

File: src/ngraph/op/fused/scatter_nd.cpp [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+{
+    element::Type data_et = get_input_element_type(DATA);
+    element::Type indices_et = get_input_element_type(INDICES);
+    element::Type updates_et = get_input_element_type(UPDATES);
+
+    const PartialShape& data_shape = get_input_shape(DATA);
```
Comment (tomdol) [inline src/ngraph/op/fused/scatter_nd.cpp]: You should use `get_input_partial_shape` instead

File: src/ngraph/op/fused/scatter_nd.cpp [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+#include "ngraph/op/scatter_nd_add.hpp"
+#include "ngraph/op/select.hpp"
+#include "ngraph/shape.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+static int DATA = 0;
+static int INDICES = 1;
+static int UPDATES = 2;
```
Comment (tsocha) [inline src/ngraph/op/fused/scatter_nd.cpp]: \+ const

File: src/ngraph/op/fused/scatter_nd.cpp [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    element::Type indices_et = get_input_element_type(INDICES);
+    element::Type updates_et = get_input_element_type(UPDATES);
+
+    const PartialShape& data_shape = get_input_shape(DATA);
+    const PartialShape& indices_shape = get_input_shape(INDICES);
+    const PartialShape& updates_shape = get_input_shape(UPDATES);
+
+    const size_t data_rank = static_cast<size_t>(data_shape.rank());
+    const size_t indices_rank = static_cast<size_t>(indices_shape.rank());
+    const size_t updates_rank = static_cast<size_t>(updates_shape.rank());
```
Comment (tsocha) [inline src/ngraph/op/fused/scatter_nd.cpp]: If rank is dynamic you will get an exception

File: src/ngraph/op/fused/scatter_nd.cpp [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+    const size_t indices_last_dim =
+        static_cast<size_t>(indices_shape[static_cast<size_t>(indices_shape.rank()) - 1]);
```
Comment (tsocha) [inline src/ngraph/op/fused/scatter_nd.cpp]: Fully dynamic shapes will cause problems here.

File: src/ngraph/op/fused/scatter_nd.hpp [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+            {
+            public:
+                static constexpr NodeTypeInfo type_info{"ScatterND", 1};
```
Comment (tsocha) [inline src/ngraph/op/fused/scatter_nd.hpp]: There is a mismatch between namespace and type_info
```suggestion
                static constexpr NodeTypeInfo type_info{"ScatterND", 0};
```

File: src/ngraph/frontend/onnx_import/op/scatter_nd.cpp [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <memory>
+
+#include "ngraph/op/fused/scatter_nd.hpp"
+#include "utils/common.hpp"
```
Comment (mitruska) [inline src/ngraph/frontend/onnx_import/op/scatter_nd.cpp]: I think you don't need `#include "utils/common.hpp"` but you should include your `scatter_nd.hpp` file here.

File: src/ngraph/op/fused/scatter_nd.hpp (hunk 1 above)
Comment (Ewa21) [reply src/ngraph/op/fused/scatter_nd.hpp]: Done

File: src/ngraph/op/fused/scatter_nd.cpp (hunk 2 above)
Comment (Ewa21) [reply src/ngraph/op/fused/scatter_nd.cpp]: Done

File: src/ngraph/op/fused/scatter_nd.cpp (hunk 5 above)
Comment (Ewa21) [reply src/ngraph/op/fused/scatter_nd.cpp]: Changed

File: src/ngraph/op/fused/scatter_nd.cpp (hunk 3 above)
Comment (Ewa21) [reply src/ngraph/op/fused/scatter_nd.cpp]: Done

File: src/ngraph/op/fused/scatter_nd.cpp (hunk 6 above)
Comment (Ewa21) [reply src/ngraph/op/fused/scatter_nd.cpp]: changed

File: src/ngraph/op/fused/scatter_nd.cpp (hunk 4 above)
Comment (Ewa21) [reply src/ngraph/op/fused/scatter_nd.cpp]: Done

File: src/ngraph/frontend/onnx_import/op/scatter_nd.cpp (hunk 8 above)
Comment (Ewa21) [reply src/ngraph/frontend/onnx_import/op/scatter_nd.cpp]: Done

File: src/ngraph/op/fused/scatter_nd.cpp [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+#include "ngraph/op/constant.hpp"
+#include "ngraph/op/convert.hpp"
+#include "ngraph/op/scatter_nd_add.hpp"
+#include "ngraph/op/select.hpp"
+#include "ngraph/shape.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+static int DATA = 0;
```
Comment (postrational) [inline src/ngraph/op/fused/scatter_nd.cpp]: The order of these imputs could change in `v1`, so placing them in the global scope is not a good idea.

File: src/ngraph/op/fused/scatter_nd.cpp [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+    set_output_type(0, data_et, data_ps);
+}
+
+NodeVector op::ScatterND::decompose_op() const
+{
+    const auto data = input_value(DATA);
+    const auto indices = input_value(INDICES);
+    const auto updates = input_value(UPDATES);
+
+    const Shape& data_shape = get_input_shape(DATA);
```
Comment (postrational) [inline src/ngraph/op/fused/scatter_nd.cpp]: Will be deprecated. Please replace with `node->input(i).get_shape()`

File: src/ngraph/op/fused/scatter_nd.cpp [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+    const Shape& updates_shape = get_input_shape(UPDATES);
+
+    element::Type data_et = get_input_element_type(DATA);
```
Comment (postrational) [inline src/ngraph/op/fused/scatter_nd.cpp]: Will be deprecated. Please replace with `node->input(i).get_element_type()`

File: src/ngraph/op/fused/scatter_nd.cpp [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+
+    const auto true_values = op::Constant::create(element::i64, updates_shape, {1});
+    const auto false_values = op::Constant::create(element::i64, data_shape, {0});
+
+    const auto mask = std::make_shared<op::v0::ScatterNDAdd>(false_values, indices, true_values);
+
+    const auto mask_bool = std::make_shared<op::v0::Convert>(mask, element::boolean);
```
Comment (postrational) [inline src/ngraph/op/fused/scatter_nd.cpp]: Please add a comment to explain what this block of code produces.

File: src/ngraph/op/fused/scatter_nd.cpp [hunk 13]
```diff
@@ -0,0 +1,133 @@
...
+
+    const auto zeros = op::Constant::create(data_et, data_shape, {0});
+
+    const auto intermediate = std::make_shared<op::v0::Select>(mask_bool, zeros, data);
```
Comment (postrational) [inline src/ngraph/op/fused/scatter_nd.cpp]: Please add another comment to explain what these 2 lines produce.

File: src/ngraph/op/fused/scatter_nd.cpp (hunk 4 above)
Comment (Ewa21) [reply src/ngraph/op/fused/scatter_nd.cpp]: It was done locally and now is on github.

File: src/ngraph/op/fused/scatter_nd.cpp (hunk 9 above)
Comment (Ewa21) [reply src/ngraph/op/fused/scatter_nd.cpp]: Changed

File: src/ngraph/op/fused/scatter_nd.cpp (hunk 11 above)
Comment (Ewa21) [reply src/ngraph/op/fused/scatter_nd.cpp]: Done

File: src/ngraph/op/fused/scatter_nd.cpp (hunk 12 above)
Comment (Ewa21) [reply src/ngraph/op/fused/scatter_nd.cpp]: Done

File: src/ngraph/op/fused/scatter_nd.cpp (hunk 10 above)
Comment (Ewa21) [reply src/ngraph/op/fused/scatter_nd.cpp]: Done

**Pull request merged**